### PR TITLE
fix: force user specify at least one pk when create debezium source

### LINKED ISSUE #5248: bug: no primary key in materialized source results in panic
### Describe the bug

When a materialized source without primary key receives a kafka `UPDATE` message [this](https://github.com/risingwavelabs/risingwave/blob/main/src/storage/src/table/streaming_table/state_table.rs#L499-L501) assertion fails.

### To Reproduce

```
// start cluster
$ ./risedev kafka

// start kafka topic and kafka producer with this specific (UPDATE) message
$ .risingwave/bin/kafka/bin/kafka-topics.sh --create --topic my-topic --bootstrap-server 127.0.0.1:29092
$ .risingwave/bin/kafka/bin/kafka-console-producer.sh --topic my-topic --bootstrap-server 127.0.0.1:29092
>{"schema":{},"payload":{"before":{"id":1,"status":null},"after":{"id":1,"status":"COMPLETED"},"source":{"version":"1.4.2.Final","connector":"postgresql","name":"postgres","ts_ms":1662716476390,"snapshot":"false","db":"simple_db","schema":"public","table":"t","txId":490,"lsn":23871176,"xmin":null},"op":"u","ts_ms":1662716476717,"transaction":null}}

// connect to postgres
$ psql -h localhost -p 4566 -d dev -U root

// Create materialized source from my-topic
dev=> create materialized source simple (id int, status varchar) with ( connector = 'kafka', topic = 'my-topic', properties.bootstrap.server = '127.0.0.1:29092', scan.startup.mode = 'earliest' ) row format DEBEZIUM_JSON;
```

stack trace:

```
thread 'risingwave-streaming-actor' panicked at 'deleting an non-existing row', /home/nstabel/Workspace/risingwave-dev/src/storage/src/table/streaming_table/state_table.rs:501:25
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:142:14
   2: risingwave_storage::table::streaming_table::state_table::StateTable<S>::batch_write_rows::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:501:25
   3: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   4: risingwave_storage::table::streaming_table::state_table::StateTable<S>::commit::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:456:52
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   6: risingwave_stream::executor::mview::materialize::MaterializeExecutor<S>::execute_inner::{{closure}}
             at ./src/stream/src/executor/mview/materialize.rs:124:59
   7: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
   8: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
   9: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
  10: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/stream/stream/mod.rs:1626:9
  11: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/stream/stream/next.rs:32:9
  12: <minitrace::future::InSpan<T> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/minitrace-0.4.0/src/future.rs:118:19
  13: risingwave_stream::executor::wrapper::trace::trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:57:60
  14: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  15: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
  16: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/stream/stream/mod.rs:1626:9
  17: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/stream/stream/next.rs:32:9
  18: <async_stack_trace::StackTraced<F> as core::future::future::Future>::poll
             at ./src/utils/async_stack_trace/src/lib.rs:140:36
  19: <futures_util::future::future::fuse::Fuse<Fut> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/future/future/fuse.rs:86:37
  20: risingwave_stream::executor::wrapper::trace::stack_trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:120:70
  21: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  22: risingwave_stream::executor::wrapper::schema_check::schema_check::{{closure}}
             at ./src/stream/src/executor/wrapper/schema_check.rs:24:1
  23: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  24: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
  25: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/stream/stream/mod.rs:1626:9
  26: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/stream/stream/next.rs:32:9
  27: risingwave_stream::executor::wrapper::epoch_check::epoch_check::{{closure}}
             at ./src/stream/src/executor/wrapper/epoch_check.rs:31:44
  28: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  29: risingwave_stream::executor::wrapper::update_check::update_check::{{closure}}
             at ./src/stream/src/executor/wrapper/update_check.rs:27:1
  30: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  31: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
  32: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
  33: <risingwave_stream::executor::dispatch::DispatchExecutor as risingwave_stream::executor::StreamConsumer>::execute::{{closure}}
             at ./src/stream/src/executor/dispatch.rs:255:9
  34: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  35: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:120:9
  36: <&mut S as futures_core::stream::Stream>::poll_next
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.23/src/stream.rs:104:9
  37: <tokio_stream::stream_ext::next::Next<St> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-stream-0.1.9/src/stream_ext/next.rs:42:9
  38: <minitrace::future::InSpan<T> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/minitrace-0.4.0/src/future.rs:118:19
  39: <async_stack_trace::StackTraced<F> as core::future::future::Future>::poll
             at ./src/utils/async_stack_trace/src/lib.rs:140:36
  40: <futures_util::future::future::fuse::Fuse<Fut> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.23/src/future/future/fuse.rs:86:37
  41: risingwave_stream::executor::actor::Actor<C>::run::{{closure}}
             at ./src/stream/src/executor/actor.rs:111:16
  42: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
  43: risingwave_stream::task::stream_manager::LocalStreamManagerCore::build_actors::{{closure}}
             at ./src/stream/src/task/stream_manager.rs:608:32
  44: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
  45: <risingwave_stream::task::stream_manager::LocalStreamManagerCore::build_actors::__Enum15994158506141524639<__Variant0,__Variant1> as core::future::future::Future>::poll
             at ./src/stream/src/task/stream_manager.rs:610:17
  46: <tokio_metrics::task::Instrumented<T> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-metrics-0.1.0/src/task.rs:2365:19
  47: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.36/src/instrument.rs:272:9
  48: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/core.rs:165:17
  49: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/loom/std/unsafe_cell.rs:14:9
  50: tokio::runtime::task::core::CoreStage<T>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/core.rs:155:13
  51: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:480:19
  52: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panic/unwind_safe.rs:271:9
  53: std::panicking::try::do_call
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:492:40
  54: __rust_try
  55: std::panicking::try
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:456:19
  56: std::panic::catch_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panic.rs:137:14
  57: tokio::runtime::task::harness::poll_future
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:468:18
  58: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:104:27
  59: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:57:15
  60: tokio::runtime::task::raw::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/raw.rs:144:5
  61: tokio::runtime::task::raw::RawTask::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/raw.rs:84:18
  62: tokio::runtime::task::LocalNotified<S>::run
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/mod.rs:381:9
  63: tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/thread_pool/worker.rs:435:13
  64: tokio::coop::with_budget::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/coop.rs:102:9
  65: std::thread::local::LocalKey<T>::try_with
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/thread/local.rs:445:16
  66: std::thread::local::LocalKey<T>::with
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/thread/local.rs:421:9
  67: tokio::coop::with_budget
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/coop.rs:95:5
  68: tokio::coop::budget
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/coop.rs:72:5
  69: tokio::runtime::thread_pool::worker::Context::run_task
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/thread_pool/worker.rs:434:9
  70: tokio::runtime::thread_pool::worker::Context::run
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/thread_pool/worker.rs:408:24
  71: tokio::runtime::thread_pool::worker::run::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/thread_pool/worker.rs:386:17
  72: tokio::macros::scoped_tls::ScopedKey<T>::set
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/macros/scoped_tls.rs:61:9
  73: tokio::runtime::thread_pool::worker::run
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/thread_pool/worker.rs:383:5
  74: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/thread_pool/worker.rs:362:45
  75: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/blocking/task.rs:42:21
  76: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.36/src/instrument.rs:272:9
  77: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/core.rs:165:17
  78: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/loom/std/unsafe_cell.rs:14:9
  79: tokio::runtime::task::core::CoreStage<T>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/core.rs:155:13
  80: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:480:19
  81: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panic/unwind_safe.rs:271:9
  82: std::panicking::try::do_call
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:492:40
  83: __rust_try
  84: std::panicking::try
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:456:19
  85: std::panic::catch_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panic.rs:137:14
  86: tokio::runtime::task::harness::poll_future
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:468:18
  87: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:104:27
  88: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/harness.rs:57:15
  89: tokio::runtime::task::raw::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/raw.rs:144:5
  90: tokio::runtime::task::raw::RawTask::poll
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/raw.rs:84:18
  91: tokio::runtime::task::UnownedTask<S>::run
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/task/mod.rs:418:9
  92: tokio::runtime::blocking::pool::Task::run
             at /home/nstabel/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.20.1/src/runtime/blocking/pool.rs:91:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Expected behavior

```
dev=> create materialized source simple (id int, status varchar) with ( coector = 'kafka', topic = 'my-topic', properties.bootstrap.server = '127.0.0.1:29092', scan.startup.mode = 'earliest' ) row format DEBEZIUM_JSON;
ERROR:  QueryError: Error message
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
Signed-off-by: tabVersion <tabvision@bupt.icu>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

as title, add a pk check in the frontend, will reject req if pk not provided

```
ERROR:  QueryError: Protocol error: Primary key must be specified when creating source with row format debezium.
```

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

~~- [ ] I have written necessary rustdoc comments~~
~~- [ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

fix #5248

### CODE REVIEW

File: src/frontend/src/handler/create_source.rs [hunk 1]
```diff
@@ -116,6 +117,19 @@ pub async fn handle_create_source(
...
+        for cons in &stmt.constraints {
+            if let TableConstraint::Unique { is_primary, .. } = cons {
+                if is_primary == &true {
+                    have_pk = true;
+                    break;
+                }
+            }
+        }
+        have_pk
+    };
```
Comment (nanderstabel) [inline src/frontend/src/handler/create_source.rs:131]:
```suggestion
    let has_pk = stmt.constraints.iter().any(|cons| {
        match cons {
            TableConstraint::Unique { is_primary, .. } => *is_primary,
            _ => false
        }
    });
```

**Comment by nanderstabel** [comment]
Should we add this to `e2e_test/source/basic_test.slt`?:
```
statement error
create materialized source no_pk (
    id integer, 
    status varchar
) with ( 
    connector = 'kafka', 
    topic = 'no_pk', 
    properties.bootstrap.server = '127.0.0.1:29092', 
    scan.startup.mode = 'earliest'
) row format debezium_json;
```
And this to `scripts/source/test_data/`?
```
{"schema":{},"payload":{"before":{"id":1,"status":null},"after":{"id":1,"status":"COMPLETED"},"source":{"version":"1.4.2.Final","connector":"postgresql","name":"postgres","ts_ms":1662716476390,"snapshot":"false","db":"simple_db","schema":"public","table":"t","txId":490,"lsn":23871176,"xmin":null},"op":"u","ts_ms":1662716476717,"transaction":null}}
```

**Comment by tabVersion** [comment]
> Should we add this to `e2e_test/source/basic_test.slt`?:
> 
> ```
> statement error
> create materialized source no_pk (
>     id integer, 
>     status varchar
> ) with ( 
>     connector = 'kafka', 
>     topic = 'no_pk', 
>     properties.bootstrap.server = '127.0.0.1:29092', 
>     scan.startup.mode = 'earliest'
> ) row format debezium_json;
> ```
> 
> And this to `scripts/source/test_data/`?
> 
> ```
> {"schema":{},"payload":{"before":{"id":1,"status":null},"after":{"id":1,"status":"COMPLETED"},"source":{"version":"1.4.2.Final","connector":"postgresql","name":"postgres","ts_ms":1662716476390,"snapshot":"false","db":"simple_db","schema":"public","table":"t","txId":490,"lsn":23871176,"xmin":null},"op":"u","ts_ms":1662716476717,"transaction":null}}
> ```

we already have a debezium related test in `scripts/source/test_data/debezium_log.1`. No need to setup a new topic for this.

**Review by nanderstabel — APPROVED** [review]
LGTM!

File: src/frontend/src/handler/create_source.rs [hunk 2]
```diff
@@ -116,6 +117,11 @@ pub async fn handle_create_source(
     is_materialized: bool,
     stmt: CreateSourceStatement,
 ) -> Result<PgResponse> {
+    let has_pk = stmt.constraints.iter().any(|cons| match cons {
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_source.rs:120]: The primary key can also be specified with column constraints. May check `row_id_column` for whether user pk exists.

**Review by BugenZhao — CHANGES_REQUESTED** [review]

File: src/frontend/src/handler/create_source.rs [hunk 3]
```diff
@@ -118,7 +119,13 @@ pub async fn handle_create_source(
...
     let (column_descs, pk_column_id_from_columns) = bind_sql_columns(stmt.columns)?;
     let (mut columns, pk_column_ids, row_id_index) =
-        bind_sql_table_constraints(column_descs, pk_column_id_from_columns, stmt.constraints)?;
+        bind_sql_table_constraints(column_descs, pk_column_id_from_columns, &stmt.constraints)?;
+
+    let has_pk = row_id_index.is_none()
+        || stmt.constraints.iter().any(|cons| match cons {
+            TableConstraint::Unique { is_primary, .. } => *is_primary,
+            _ => false,
+        });
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_source.rs:128]: There seems no need to check table constraints any more. It has been resolved in `bind_sql_table_constraints`.

File: src/frontend/src/handler/create_source.rs [hunk 4]
```diff
@@ -118,7 +119,13 @@ pub async fn handle_create_source(
 ) -> Result<PgResponse> {
     let (column_descs, pk_column_id_from_columns) = bind_sql_columns(stmt.columns)?;
     let (mut columns, pk_column_ids, row_id_index) =
-        bind_sql_table_constraints(column_descs, pk_column_id_from_columns, stmt.constraints)?;
+        bind_sql_table_constraints(column_descs, pk_column_id_from_columns, &stmt.constraints)?;
+
+    let has_pk = row_id_index.is_none()
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_source.rs:124]:
```suggestion
    let has_user_specified_pk = row_id_index.is_none()
```

**Review by xiangjinwu — APPROVED** [review]
Just fix the unused `TableConstraint` warning

**Review by BugenZhao — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]

**Pull request merged**